configure.ac

```diff
@@ -192,17 +192,22 @@ AS_IF([test "$enable_long_fortran_header" = "yes"], [FORTRAN_LONG_HEADER=" "], [
 AC_SUBST([FORTRAN_LONG_HEADER])
 AM_CONDITIONAL([HAVE_LONG_FORTRAN_HEADER], [test "$enable_long_fortran_header" = "yes"])
 
-AC_ARG_ENABLE([mr-scalable],
-    [AC_HELP_STRING([--disable-mr-scalable],
-                    [Disable the use of scalable memory regions in the OFI transport.  Required by some providers. (default: enabled)])])
-AS_IF([test "$enable_mr_scalable" != "no"],
-      [AC_DEFINE([ENABLE_MR_SCALABLE], [1], [If defined, the OFI transport will use FI_MR_SCALABLE.  Otherwise, FI_MR_BASIC is used.])])
+AC_ARG_ENABLE([ofi-mr],
+    [AC_HELP_STRING([--enable-ofi-mr=MODE],
+                    [OFI memory registration mode: basic, scalable, or rma-event (default: scalable)])])
+
+AS_IF([test -z "$enable_ofi_mr"], [enable_ofi_mr="scalable"])
+
+AS_CASE([$enable_ofi_mr],
+        [basic],
+            [],
+        [scalable],
+            [AC_DEFINE([ENABLE_MR_SCALABLE], [1], [If defined, the OFI transport will use FI_MR_SCALABLE])],
+        [rma*event],
```

Comment:
Is this a wildcard?

Reply:
Should we document this new flag behavior in the README?

Reply:
You bet it is!  I was thinking it might be good to also accept 'rma_event'.  But, I don't mind restricting this, if you think it would be better.

It's helpful to document the basic/required options in the README file, but I prefer not to document all of the configure options.  Partially because it's a pain to keep such things in sync.  The README says "See 'configure --help' for documentation on available options."  To that end, there's a bunch of stuff we could delete from the README...

Reply:
Ah yes, the underscore.  That makes sense.

I don't think it matters much at all in this case, but out of curiosity, is it possible to do an m4 regex here for "-" or "_"?

Agreed on the README.

Reply:
No, it's m4sh, so this translates into a shell case statement.  It might be portable to use `?` instead of `*` for a single character wildcard.  I'll look into that.